waspc/data/Generator/templates/server/src/jobs/core/pgBoss/pgBossJob.js

```diff
@@ -51,14 +51,15 @@ class PgBossJob extends Job {
   }
 
   /**
-   * Submits the job to PgBoss.
+   * Submits the job to pg-boss.
    * @param {object} jobArgs - The job arguments supplied by the user for their perform callback.
-   * @param {object} jobOptions - PgBoss specific options for `boss.send()`, which can override their defaultJobOptions.
+   * @param {object} jobOptions - pg-boss specific options for `boss.send()`, which can override their defaultJobOptions.
    */
   async submit(jobArgs, jobOptions) {
+    const boss = await pgBossStarted
```

Comment:
Good observation! I originally wanted to do the same. Unfortunately, the `PgBossJob` instance is created and returned from `createJob`. This is actually called at the module level in each Job JS file, but _before_ pg-boss itself starts. Attempting to put that line in that function causes NodeJS to crash, presumably because the system cannot resolve all the modules before starting the server, which starts pg-boss? Would love to hop on a call to brainstorm ideas if you have time though!

Reply:
We talked about this in person (well, over video) and decided to not to do anything about it for now. I'll take a look at whether we can use async imports and top-level await to fix it (added a TODO for myself, will make an issue).